src/NuGet.Core/NuGet.Packaging/Signing/Timestamp/Rfc3161TimestampProvider.cs

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography;
+
+#if IS_DESKTOP
+using System.Security.Cryptography.Pkcs;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+#endif
+
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Common;
+
+namespace NuGet.Packaging.Signing
+{
+    /// <summary>
+    /// A provider for RFC 3161 timestamps
+    /// https://tools.ietf.org/html/rfc3161
+    /// </summary>
+    public class Rfc3161TimestampProvider : ITimestampProvider
+    {
+        // Url to an RFC 3161 timestamp server
+        private readonly Uri _timestamperUrl;
+        private const int _rfc3161RequestTimeoutSeconds = 10;        
+
+        public Rfc3161TimestampProvider(Uri timeStampServerUrl)
+        {
+#if IS_DESKTOP
+            // Uri.UriSchemeHttp and Uri.UriSchemeHttps are not available in netstandard 1.3
+            if (!string.Equals(timeStampServerUrl.Scheme, Uri.UriSchemeHttp, StringComparison.Ordinal) &&
+                !string.Equals(timeStampServerUrl.Scheme, Uri.UriSchemeHttps, StringComparison.Ordinal))
```

Comment:
is it possible for the case to be off here?

Reply:
I dont think so, the Scheme property cannot be set - https://msdn.microsoft.com/en-us/library/system.uri.scheme(v=vs.110).aspx